src/evm_parse.c

```diff
@@ -418,6 +475,223 @@ enum parse_rv parse_rlp_txn(struct EVM_RLP_list_state *const state, evm_parser_m
+          if(first < 0xf8) {
+              state->remaining = first - 0xc0;
+              state->state=2;
+          } else {
+              state->len_len = first - 0xf7;
+              state->state=1;
+          }
+      }
+      case 1:
+        if(state->state==1) {
+            // Max length we could get for this value is 8 bytes so uint64_state is appropriate.
+            sub_rv = parseFixed(((struct FixedState*)&state->uint64_state), &meta->input, state->len_len);
+            if(sub_rv != PARSE_RV_DONE) return sub_rv;
+            for(size_t i = 0; i < state->len_len; i++) {
+                ((uint8_t*)(&state->remaining))[i] = state->uint64_state.buf[state->len_len-i-1];
+            }
+        }
+        init_rlp_item(&state->rlpItem_state);
+        state->state = 2;
+      case 2: { // Now parse items.
+          uint8_t itemStartIdx;
+          switch(state->item_index) {
+            PARSE_ITEM(EVM_EIP1559_TXN_CHAINID, _to_buffer);
+
+            if(state->rlpItem_state.length != 2
+               || state->rlpItem_state.buffer[0] != 0xa8
+               || (state->rlpItem_state.buffer[1] != 0x68
+                   && state->rlpItem_state.buffer[1] != 0x69
+                   && state->rlpItem_state.buffer[1] != 0x6a))
+                REJECT("Chain ID incorrect for the Avalanche C chain");
+            meta->chainIdLowByte = 0; // explicitly clear chain ID low byte for EIP1559 transactions - only legacy transactions needed to include it
+
+            FINISH_ITEM_CHUNK();
+
+            PARSE_ITEM(EVM_EIP1559_TXN_NONCE, );
+            FINISH_ITEM_CHUNK();
+
+            PARSE_ITEM(EVM_EIP1559_TXN_MAX_PRIORITY_FEE_PER_GAS, _to_buffer);
+            uint64_t maxFeePerGas = enforceParsedScalarFits64Bits(&state->rlpItem_state);
+            size_t maxFeePerGasLength = state->rlpItem_state.length;
+            state->priorityFeePerGas = maxFeePerGas;
+            FINISH_ITEM_CHUNK();
+
+            PARSE_ITEM(EVM_EIP1559_TXN_MAX_FEE_PER_GAS, _to_buffer);
+            uint64_t baseFeePerGas = enforceParsedScalarFits64Bits(&state->rlpItem_state);
+            size_t baseFeePerGasLength = state->rlpItem_state.length;
+            state->baseFeePerGas = baseFeePerGas;
+            FINISH_ITEM_CHUNK();
+
+            PARSE_ITEM(EVM_EIP1559_TXN_GAS_LIMIT, _to_buffer);
+            uint64_t gasLimit = enforceParsedScalarFits64Bits(&state->rlpItem_state);
+            size_t gasLimitLength = state->rlpItem_state.length;
+            state->gasLimit = gasLimit;
+            FINISH_ITEM_CHUNK();
+
+            // TODO: We don't currently support the C-chain gas limit of 100 million,
+            // which would have a fee larger than what fits in a word
+            // possible future improvement: using nanosdk to actually do the overflow-checked gas arithmetic
+            if((maxFeePerGasLength > baseFeePerGasLength ? maxFeePerGasLength : baseFeePerGasLength) + gasLimitLength + 1 > 8)
```

Comment:
FWIW there's a `MAX` macro in `types.h`

Reply:
Why `x + 1 > 8`? 
That means `x >= 8` but I'd expect `x > 8` to be the length-based overflow check.